fast_response/MonitoringAndMocks/Data_Display.py

```diff
@@ -8,756 +8,387 @@
+    ###create empty list and load each value into the list which is then loaded into the dict.
+    loading_dict = {"Trigger_Time": [], "GCN_Alert": [], "End_Time": [],
+                "LVK_Latency": [], "IceCube_Latency": [], "Total_Latency": [], 
+                "Name": [], "time_stamp": [], "first": [], 
+                "last": [], "we_had_to_wait": []}
+    
+    for i in mock_files:
+        mocks = pd.read_pickle(f'{i}')
+        ###load data from pickle files to dataframe
+        for key in select_keys:
+            mock_key = "Ligo_Latency" if key == "LVK_Latency" else key
+            loading_dict[key] = mocks[mock_key]
+
+        ###separate name of event
+        ###file name format: gw_latency_dict_S240501dr-2-Preliminary_test.pickle
+        first_split = i.split('_')
+        second_split = first_split[3].split('-')
+        loading_dict["Name"] = second_split[0]
+
+        ###collect timestamps of creation for mock files
+        timestamp = os.path.getmtime(i)
+        datestamp = datetime.datetime.fromtimestamp(timestamp, datetime.UTC)
+        loading_dict["time_stamp"] = datestamp
+
+        ###is it the first or last file sent? not all events have 3 files,
+        ###so determinging "last" will require more work in another function below
+        if '-1-' in str(i):
+            loading_dict["first"] = True
+            loading_dict["last"] = False
+        else:
+            loading_dict["first"] = False
+        if '-3-' in str(i):
+            loading_dict["last"] = True
+        if '-2-' in str(i):
+            if len(glob.glob(path + f'PickledMocks/*{second_split[0]}-3-*'.format())) > 0:
+                loading_dict["last"] = False
             else:
-                all_dictionary[Key].append(Entries[Key])
-        all_dictionary['Name'].append(file)
-    i+=1
-
-for file in all_dictionary["Name"]:
-        date_format = "%Y-%m-%d"
-        c_timestamp = os.path.getmtime(file)
-        c_datestamp = datetime.datetime.utcfromtimestamp(c_timestamp)
-        all_dictionary['Time_Stamp'].append(c_datestamp)
-print('Finished loading all latencies.')
-
-#Setting conditions for call to be made if script fails. Condition: if more than 3 hours pass between alerts
-if (Time(datetime.datetime.utcnow()).mjd - max(Time(all_dictionary["Time_Stamp"]).mjd)) > 3600.*3/86400:
+                loading_dict["last"] = True
+        ###500 seconds of data must be collected before i3 can run, anything more is delay
+        delay = 500 - mocks["Ligo_Latency"]
+        loading_dict["we_had_to_wait"] = delay
+        event_dict = event_dict.append(loading_dict, ignore_index=True)
+    return event_dict
+
+event_dict = sort_mocks(mock_files=mock_files)
+
+###now to make the plots###
+ed = event_dict
+
+###call function incase mocks get interrupted
+if (Time(datetime.datetime.now(timezone.utc)).mjd - max(Time(ed["Time_Stamp"]).mjd)) > 3600.*3/86400:
+        print(datetime.datetime.now(timezone.utc))
```

Comment:
you can probably remove these two lines (print statements on 97/98), the message on 101 is sufficient